### PR TITLE
[AMORO-1875]Support display paimon files in 'files' page

### LINKED ISSUE #1875: [Subtask]: Support display paimon files in 'files' page
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

#1269 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shidayang**
## Why are the changes needed?
Close #1875.

## Brief change log
- Implement 'getTablePartition' and 'getTableFile' method of PaimonTableDescriptor

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by shidayang** [comment]
image
image

**Review by baiyangtx — COMMENTED** [review]
Should we add some unit tests for this section?

**Review by baiyangtx — COMMENTED** [review]
It is necessary to display the Commit ID-related information. If it cannot be obtained, it can be displayed as empty.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java [hunk 1]
```diff
@@ -154,12 +162,58 @@ public List<DDLInfo> getTableOperations(AmoroTable<?> amoroTable) {
 
   @Override
   public List<PartitionBaseInfo> getTablePartition(AmoroTable<?> amoroTable) {
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:164]: `getTablePartitions` may be a better method name.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java [hunk 2]
```diff
@@ -154,12 +162,58 @@ public List<DDLInfo> getTableOperations(AmoroTable<?> amoroTable) {
...
+          lastCommitTime = Math.max(lastCommitTime, dataFileMeta.creationTime().getMillisecond());
+        }
+        partitionBaseInfoList.add(new PartitionBaseInfo(partitionSt, fileCount, fileSize, lastCommitTime));
+      }
+    }
+    return partitionBaseInfoList;
   }
 
   @Override
   public List<PartitionFileBaseInfo> getTableFile(AmoroTable<?> amoroTable, String partition) {
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:193]: `getTableFiles` may be a better method name.

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:193]: BTW, we lost some `@Override` for methods in `com.netease.arctic.server.dashboard.MixedAndIcebergTableDescriptor`.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java [hunk 3]
```diff
@@ -154,12 +162,58 @@ public List<DDLInfo> getTableOperations(AmoroTable<?> amoroTable) {
...
-    throw new UnsupportedOperationException();
+    FileStoreTable table = getTable(amoroTable);
+    AbstractFileStore<?> store = (AbstractFileStore<?>) table.store();
+    FileStorePathFactory fileStorePathFactory = store.pathFactory();
+    List<ManifestEntry> files = store.newScan().plan().files(FileKind.ADD);
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:197]: It seems that partition filtering is missing here.

**Review by wangtaohz — COMMENTED** [review]
I left some comment, please take a look @shidayang

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java [hunk 4]
```diff
@@ -154,12 +162,58 @@ public List<DDLInfo> getTableOperations(AmoroTable<?> amoroTable) {
...
+
+    List<PartitionBaseInfo> partitionBaseInfoList = new ArrayList<>();
+    for (Map.Entry<BinaryRow, Map<Integer, List<DataFileMeta>>> groupByPartitionEntry : groupByPartFiles.entrySet()) {
+      for (Map.Entry<Integer, List<DataFileMeta>> groupByBucketEntry : groupByPartitionEntry.getValue().entrySet()) {
+        String partitionSt = partitionString(
+            groupByPartitionEntry.getKey(),
+            groupByBucketEntry.getKey(),
+            fileStorePathFactory);
+        int fileCount = 0;
+        int fileSize = 0;
```
Comment (HuangFru) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:179]: 'long' to avoid overflow.

**Review by HuangFru — CHANGES_REQUESTED** [review]
image
Same here.

**Comment by shidayang** [comment]
> Should we add some unit tests for this section?

Will add in #2119

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java (hunk 1 above)
Comment (shidayang) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:164]: yeah

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java (hunk 2 above)
Comment (shidayang) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/PaimonTableDescriptor.java:193]: yeah

**Review by wangtaohz — APPROVED** [review]
LGTM

**Review by HuangFru — APPROVED** [review]
LGTM.

**Comment by baiyangtx** [comment]
Cloud you like to update the screenshot

**Comment by shidayang** [comment]
> It is necessary to display the Commit ID-related information. If it cannot be obtained, it can be displayed as empty.

image
